time/time.c

```diff
@@ -91,26 +92,29 @@ time_t time(time_t *tp)
 
 int clock_gettime(clockid_t clk_id, struct timespec *tp)
 {
-	time_t now, offs;
+	time_t time, offs;
 
 	if (tp == NULL) {
 		errno = EINVAL;
 		return -1;
 	}
 
-	if (clk_id != CLOCK_REALTIME && clk_id != CLOCK_MONOTONIC && clk_id != CLOCK_MONOTONIC_RAW) {
+	if (clk_id != CLOCK_REALTIME && clk_id != CLOCK_MONOTONIC && clk_id != CLOCK_MONOTONIC_RAW && clk_id != CLOCK_THREAD_CPUTIME_ID) {
 		errno = EINVAL;
 		return -1;
 	}
 
-	gettime(&now, &offs);
-
-	if (clk_id == CLOCK_REALTIME)
-		now += offs;
+	if (clk_id == CLOCK_THREAD_CPUTIME_ID)
+		time = (unsigned int)getThreadCpuTime();
```

Comment:
The body of if statement should be in brackets. 
MISRA: `Rule 15.6 The body of an iteration-statement or a selection-statement shall be a compound-statement`